webui/src/pages/repositories/repository/fileRenderers/index.tsx

```diff
@@ -1,96 +1,342 @@
+    if (lang)
+        return (
+        <TextDownloader {...props} onReady={text =>
+            <TextRenderer {...props} text={text}/>}
+        />
+    );
+
+    return <UnsupportedFileType {...props}/>
+}
+
+export const ObjectTooLarge: FC<RendererComponent> = ({path, sizeBytes}) => {
+    return (
+        <Alert variant="warning" className="m-5">
+            <div>Could not render: <code>{path}</code>:</div>
+            <div>{`size ${sizeBytes}b (${humanSize(sizeBytes)}) is too big`}</div>
+        </Alert>
+    );
+}
+
+export const UnsupportedFileType: FC<RendererComponent> = ({path, fileExtension, contentType}) => {
+    return (
+        <Alert variant="warning" className="m-5">
+            <div>Could not render: <code>{path}</code>: <br/></div>
+            <div>{`lakeFS doesn't know how to render this file (extension = "${fileExtension}", content-type = "${contentType}")`}</div>
+        </Alert>
+    );
+}
+
+export const DataLoader: FC<RendererComponent> = () => {
+    return <Loading/>
+}
+
+export const TextDownloader: FC<RendererComponentWithTextCallback> = (props) => {
+    const { repoId, refId, path, onReady } = props;
+    const {response, error, loading} = useAPI(
+        async () => await objects.get(repoId, refId, path),
+        [repoId, refId, path]
+    );
+    if (loading) {
+        return <DataLoader {...props}/>
+    }
+    if (error) {
+        return <Error error={error}/>
+    }
 
-export interface GenericRendererProps {
-    content: string,
-    language: string,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const component = onReady((response as any));
+    return (
+        <>
+            {component}
+        </>
+    )
 }
 
-export const GenericRenderer: FC<GenericRendererProps> = ({content, language}) => {
+export const MarkdownRenderer: FC<RendererComponentWithText> = ({text}) => {
+    return (
+            <ReactMarkdown remarkPlugins={[remarkGfm, remarkHtml]} linkTarget={"_blank"}>
+                {text}
+            </ReactMarkdown>
+    );
+
+};
+
+export const TextRenderer: FC<RendererComponentWithText> = ({ contentType, fileExtension, text }) => {
+    const language = guessLanguage(fileExtension, contentType) ?? "plaintext";
     return (
         <SyntaxHighlighter
             style={syntaxHighlightStyle}
             language={language}
             showInlineLineNumbers={true}
-            showLineNumbers={true}>{content}</SyntaxHighlighter>
+            showLineNumbers={true}>{text}</SyntaxHighlighter>
     );
 };
 
-export const IpynbRenderer: FC<RendererComponent> = ({ content }) => {
-    if (content) {
-        return (
-            <NbRenderer
-                ipynb={JSON.parse(content)}
-                syntaxTheme="ghcolors"
-                language="python"
-                bgTransparent={true}
-            />
-        );
+
+export const DuckDBRenderer: FC<RendererComponentWithDataFormat> = (props) => {
```

Comment:
I'd move `DuckDBRenderer` to another file along with `DataRow`, as it's getting a bit long. I should probably refactor to move all renderers out of this file. I should have done it this way to begin with.

Reply:
Done